File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/dbdebug/DbDebugFragment.java
package com.didichuxing.doraemonkit.kit.dbdebug;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.text.Html;
import android.view.View;
import android.widget.TextView;

import com.amitshekhar.DebugDB;
import com.amitshekhar.debug.encrypt.sqlite.DebugDBEncryptFactory;
import com.amitshekhar.debug.sqlite.DebugDBFactory;
import com.blankj.utilcode.util.NetworkUtils;
import com.didichuxing.doraemonkit.DoraemonKit;
import com.didichuxing.doraemonkit.DoraemonKitReal;
import com.didichuxing.doraemonkit.R;
import com.didichuxing.doraemonkit.ui.base.BaseFragment;
import com.didichuxing.doraemonkit.ui.dialog.DialogInfo;
import com.didichuxing.doraemonkit.ui.dialog.SimpleDialogListener;
import com.didichuxing.doraemonkit.ui.setting.SettingItem;
import com.didichuxing.doraemonkit.ui.setting.SettingItemAdapter;
import com.didichuxing.doraemonkit.ui.widget.recyclerview.DividerItemDecoration;
import com.didichuxing.doraemonkit.ui.widget.titlebar.HomeTitleBar;
import com.didichuxing.doraemonkit.util.DataCleanUtil;

import java.util.ArrayList;
import java.util.List;

/**
 * @author jintai
 * Created by jintai on 2019/10/17.
 * 数据库远程调试介绍页面
 */

public class DbDebugFragment extends BaseFragment {
    TextView tvIp;

    @Override
    protected int onRequestLayout() {
        return R.layout.dk_fragment_db_debug;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initView();
    }

    private void initView() {
        if (!DebugDB.isServerRunning()) {
            DebugDB.initialize(DoraemonKitReal.APPLICATION, new DebugDBFactory());
            DebugDB.initialize(DoraemonKitReal.APPLICATION, new DebugDBEncryptFactory());
        }
        HomeTitleBar titleBar = findViewById(R.id.title_bar);
        titleBar.setListener(new HomeTitleBar.OnTitleBarClickListener() {
            @Override
            public void onRightClick() {
                finish();
            }
        });
        TextView tvTip = findViewById(R.id.tv_tip);
        tvTip.setText(Html.fromHtml(getResources().getString(R.string.dk_kit_db_debug_desc)));
        tvIp = findViewById(R.id.tv_ip);
        if (DebugDB.isServerRunning()) {
            tvIp.setText("" + DebugDB.getAddressLog().replace("Open ", "").replace("in your browser", ""));
        } else {
            tvIp.setText("servse is not start");
        }
    }

    /**
     * 网络变化时调用
     *
     * @param networkType
     */
    public void networkChanged(NetworkUtils.NetworkType networkType) {
        if (tvIp == null) {
            return;
        }
        if (networkType == NetworkUtils.NetworkType.NETWORK_NO) {
            tvIp.setText("please check network is connected");
        } else {
            tvIp.setText("" + DebugDB.getAddressLog().replace("Open ", "").replace("in your browser", ""));
        }
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/dbdebug/DbDebugFragment.kt
package com.didichuxing.doraemonkit.kit.dbdebug

import android.os.Bundle
import android.text.Html
import android.view.View
import android.widget.TextView
import com.amitshekhar.DebugDB
import com.amitshekhar.debug.encrypt.sqlite.DebugDBEncryptFactory
import com.amitshekhar.debug.sqlite.DebugDBFactory
import com.blankj.utilcode.util.NetworkUtils
import com.didichuxing.doraemonkit.DoraemonKit
import com.didichuxing.doraemonkit.R
import com.didichuxing.doraemonkit.kit.core.BaseFragment
import com.didichuxing.doraemonkit.widget.titlebar.HomeTitleBar

/**
 * @author jintai
 * Created by jintai on 2019/10/17.
 * 数据库远程调试介绍页面
 */
class DbDebugFragment : BaseFragment() {
    private var tvIp: TextView? = null
    override fun onRequestLayout(): Int {
        return R.layout.dk_fragment_db_debug
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        try {
            initView()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    @Throws(Exception::class)
    private fun initView() {
        if (!DebugDB.isServerRunning()) {
            DebugDB.initialize(DoraemonKit.APPLICATION, DebugDBFactory())
            DebugDB.initialize(DoraemonKit.APPLICATION, DebugDBEncryptFactory())
        }
        val titleBar = findViewById<HomeTitleBar>(R.id.title_bar)
        titleBar.setListener(object : HomeTitleBar.OnTitleBarClickListener {
            override fun onRightClick() {
                finish()
            }
        })
        val tvTip = findViewById<TextView>(R.id.tv_tip)
        tvTip.text = Html.fromHtml(resources.getString(R.string.dk_kit_db_debug_desc))
        tvIp = findViewById(R.id.tv_ip)
        if (DebugDB.isServerRunning()) {
            tvIp?.text = DebugDB.getAddressLog().replace("Open ", "").replace("in your browser", "")
        } else {
            tvIp?.text = "servse is not start"
        }
    }

    /**
     * 网络变化时调用
     *
     * @param networkType
     */
    fun networkChanged(networkType: NetworkUtils.NetworkType) {
        if (tvIp == null) {
            return
        }
        if (networkType == NetworkUtils.NetworkType.NETWORK_NO) {
            tvIp?.text = "please check network is connected"
        } else {
            tvIp?.text = DebugDB.getAddressLog().replace("Open ", "").replace("in your browser", "")
        }
    }
}